No module named 'numpy.lib.index_tricks' (numpy 2.0)

Saw it this morning over at Jdaviz using numpy nightly wheel.

```
.../photutils/__init__.py:16: in <module>
    from . import background
.../photutils/background/__init__.py:7: in <module>
    from .background_2d import *  # noqa: F401, F403
.../photutils/background/background_2d.py:15: in <module>
    from numpy.lib.index_tricks import index_exp
E   ModuleNotFoundError: No module named 'numpy.lib.index_tricks'
```

https://github.com/astropy/photutils/blob/5d8795814b07654541eaa7ddde9031c481eb246f/photutils/background/background_2d.py#L15C13-L15C13

Hard to tell from https://github.com/numpy/numpy/pull/24581 but looks like all you need to do is use `numpy.index_exp` directly? Maybe it is even backward compatible.

--- a/photutils/background/background_2d.py
+++ b/photutils/background/background_2d.py
@@ -12,7 +12,6 @@ from astropy.nddata import NDData
 from astropy.stats import SigmaClip
 from astropy.utils import lazyproperty
 from astropy.utils.exceptions import AstropyUserWarning
-from numpy.lib.index_tricks import index_exp
 
 from photutils.aperture import RectangularAperture
 from photutils.background.core import SExtractorBackground, StdBackgroundRMS
@@ -306,7 +305,7 @@ class Background2D:
                 self.nboxes = data.shape // self.box_size
             elif self.edge_method == 'crop':
                 crop_size = self.nboxes * self.box_size
-                crop_slc = index_exp[0:crop_size[0], 0:crop_size[1]]
+                crop_slc = np.index_exp[0:crop_size[0], 0:crop_size[1]]
                 data = self.data[crop_slc]
             else:
                 raise ValueError('edge_method must be "pad" or "crop"')

--- a/photutils/conftest.py
+++ b/photutils/conftest.py
@@ -3,12 +3,19 @@
 # get picked up when running the tests inside an interpreter using
 # packagename.test
 
+import numpy as np
+from astropy.utils import minversion
+
 try:
     from pytest_astropy_header.display import (PYTEST_HEADER_MODULES,
                                                TESTED_VERSIONS)
     ASTROPY_HEADER = True
 except ImportError:
     ASTROPY_HEADER = False
+
+# do not remove until we drop support for NumPy < 2.0
+if minversion(np, "2.0.0.dev0+git20230726"):
+    np.set_printoptions(legacy='1.25')
 
 
 def pytest_configure(config):

--- a/photutils/isophote/fitter.py
+++ b/photutils/isophote/fitter.py
@@ -139,7 +139,7 @@ class EllipseFitter:
         # build the resulting Isophote in cases where iterations run to
         # the maximum allowed (maxit), or the maximum number of flagged
         # data points (fflag) is reached.
-        minimum_amplitude_value = np.Inf
+        minimum_amplitude_value = np.inf
         minimum_amplitude_sample = None
 
         # these must be passed throughout the execution chain.

--- a/setup.cfg
+++ b/setup.cfg
@@ -79,6 +79,8 @@ filterwarnings =
     ignore:numpy\.ndarray size changed:RuntimeWarning
     # numpy 1.24 deprecation from skimage (can remove for skimage >= 0.20)
     ignore:`np.bool8` is a deprecated alias for `np.bool_`:DeprecationWarning
+    # TEMP: remove when matplotlib nightly fixes the row_stack deprecations
+    ignore:`row_stack` alias is deprecated:DeprecationWarning
 
 [coverage:run]
 omit =

